tests/test_grammar.py

```diff
@@ -357,13 +364,16 @@
     ("str_top_esc_inter", r"Esc: \${str}", "Esc: ${str}"),
     ("str_top_esc_inter_wrong_1", r"Wrong: $\{str\}", r"Wrong: $\{str\}"),
     ("str_top_esc_inter_wrong_2", r"Wrong: \${str\}", r"Wrong: ${str\}"),
-    ("str_top_esc_backslash", r"Esc: \\${str}", r"Esc: \hi"),
+    ("str_top_esc_backslash_1", r"Esc: \\${str}", r"Esc: \hi"),
+    ("str_top_esc_backslash_2", r"Esc: \\\\${str}", r"Esc: \\hi"),
     ("str_top_quoted_braces_wrong", r"Wrong: \{${str}\}", r"Wrong: \{hi\}"),
     ("str_top_leading_dollars", r"$$${str}", "$$hi"),
     ("str_top_trailing_dollars", r"${str}$$$$", "hi$$$$"),
-    ("str_top_leading_escapes", r"\\\\\${str}", r"\\${str}"),
-    ("str_top_middle_escapes", r"abc\\\\\${str}", r"abc\\${str}"),
-    ("str_top_trailing_escapes", "${str}" + "\\" * 5, "hi" + "\\" * 3),
+    ("str_top_leading_escapes_1", r"\\\\\${str}", r"\\${str}"),
+    ("str_top_leading_escapes_2", r"\\\\ \${str}", r"\\\\ ${str}"),
+    ("str_top_middle_escapes_1", r"abc\\\\\${str}", r"abc\\${str}"),
+    ("str_top_middle_escapes_2", r"abc\\\\ \${str}", r"abc\\\\ ${str}"),
+    ("str_top_trailing_escapes", "${str}" + "\\" * 5, "hi" + "\\" * 5),
```

Comment:
I don't think we need 5 `\\` here. this is easier to understand and testing the same thing as far as I can tell:
```suggestion
    ("str_top_trailing_escapes", r"${str}\\ ".strip(), r"hi\\".strip()),
```

Reply:
I made it 3 `\\\` in 1eb0dcc